buffer/src/main/java/io/netty/buffer/PoolThreadCache.java

```diff
@@ -231,11 +238,14 @@ boolean add(PoolArena<?> area, PoolChunk chunk, long handle, int normCapacity, S
      *  Should be called if the Thread that uses this cache is about to exist to release resources out of the cache
      */
     void free() {
-        ThreadDeathWatcher.unwatch(thread, freeTask);
-        free0();
+        Thread thread = Thread.currentThread();
+        if (freeTask != null) {
+            ThreadDeathWatcher.unwatch(thread, freeTask);
+        }
+        free0(thread);
     }
 
-    private void free0() {
+    private void free0(Thread thread) {
```

Comment:
maybe just use `Thread.currentThread()` in the log statement, and also log the `deathWatchThread` too if they may be different.